gce/burst-training/README.md

```diff
@@ -0,0 +1,327 @@
+2. The Compute Engine instance should be shut down immediately after the `python
+   censys-analysis.py` call has returned. This will keep Compute Engine charges
+   to a minimum.
+
+We use two Compute Engine mechanisms to achieve this behavior:
+
+1. [Instance
+   metadata](https://cloud.google.com/compute/docs/storing-retrieving-metadata)
+   -- These are key-value pairs of strings that you can set on a Compute Engine
+   VM. In burst training, we use them to set values for command line arguments
+   as well as to deliver the training code to the instance. The additional
+   benefit to using metadata like this is that they are visible on your
+   instances either through the Cloud Console or through `gcloud compute
+   instances describe <INSTANCE_NAME>`.
+
+2. [Startup scripts](https://cloud.google.com/compute/docs/startupscript) --
+   This is a script that is executed immediately once a Compute Engine instance
+   has started. In burst training, as we are using a Linux image (although you
+   can alter this as necessary), we use a
+   [bash](https://www.gnu.org/software/bash/) script which:
+
+   - Loads the appropriate instance metadata
+
+   - Downloads the trainer code to the VM from Google Cloud Storage
+
+   - Runs the training job specifying the appropriate command line arguments
+     from the metadata
+
+   - Shuts down the instance once the training is complete
+
+The [training script](./train.sh) shows this process in action. Once you have
+built your VM image, you can run this script to execute your first burst
+training job as follows:
+
+```
+./train.sh <TRAINING-INSTANCE-NAME> <BUCKET-NAME> <IMAGE-FAMILY>
+```
+
+Here, `<BUCKET-NAME>` and `<IMAGE-FAMILY>` should have the same values as above.
+`<TRAINING-INSTANCE-NAME>` should be a fresh name for this new training
+instance.
+
+
+#### Preemptibility
+
+The [sample training script](./train.sh) specifies that training should be
+performed on a [preemptible VM
+instance](https://cloud.google.com/compute/docs/instances/preemptible). This
+makes burst training even cheaper than it otherwise would be -- preemptible
+Compute Engine VMs can be up to 80% cheaper than regular VMs.
+
+In the case of a census sample, using a preemptible VM makes sense because, even
+at scale, the job is short-lived. If you have a longer-running training job, you
+have two options:
+
+1. Specify that you do not want to use a preemptible VM for training by removing
+   the `--preemptible` flag from `gcloud compute instances create` command in
+   the training script.
+
+2. Set your job up as per our [survival training
+   guide](https://github.com/GoogleCloudPlatform/ml-on-gcp/tree/master/gce/survival-training)
+   so that it can take advantage of preemptible VMs on Compute Engine.
+
+
+## Customization
+
+You can adapt the procedure described here to perform burst training of your own
+models. Moreover, you should be able to do so with only slight modifications to
+the code:
+
+### Burst training checklist
```

Comment:
Checklist should be repeated above in the Execution section.

Rename Customization section to "How to use this with your own code"

Reply:
Done.